src/main.ts

```diff
@@ -0,0 +1,163 @@
+import * as core from '@actions/core'
+import * as coreCommand from '@actions/core/lib/command'
+import * as fsHelper from './fs-helper'
+import * as gitSourceProvider from './git-source-provider'
+import * as github from '@actions/github'
+import * as path from 'path'
+
+async function run(): Promise<void> {
```

Comment:
this is the entry file. it just validates the inputs, and then calls the source provider which performs all of the business logic